retrieval/discovery/consul/consul.go

```diff
@@ -48,8 +49,40 @@ const (
 	datacenterLabel = model.MetaLabelPrefix + "consul_dc"
 	// serviceIDLabel is the name of the label containing the service ID.
 	serviceIDLabel = model.MetaLabelPrefix + "consul_service_id"
+
+	// Constants for instrumentation.
+	namespace = "prometheus"
 )
 
+var (
+	rpcCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "consul_sd_rpc_calls_total",
+			Help:      "Number of Consul RPC calls.",
+		},
+		[]string{"endpoint", "call"},
+	)
+	rpcFailuresCount = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "consul_sd_rpc_call_failures_total",
```

Comment:
hmm, should we do these all as sd_consul rather than consul_sd? That'll keep all the SDs together